enhancements/sig-architecture/34-work-executor-group/README.md

```diff
@@ -0,0 +1,275 @@
+explicitly check whether that "ambassador" has sufficient RBAC permission 
+in the managed cluster via SubjectAccessReview api. Due to the limit of 
+the SubjectAccessReview api, we need to send (# of contained objects) times 
+SAR requests to the spoke cluster, but this should be fine b/c the work-agent 
+is working asynchronously, caching the SAR result will be helpful for shortening
+the time cost.
+
+Note that the authorization via subject access review api doesn't require the
+service account to be actually present in the managed cluster, so in the cases
+where you have too many executor subject to manage, the number of service 
+accounts resources in the managed cluster is bounded.
+
+A sample of the requesting `SubjectAccessReview` payload is:
+
+```yaml
+apiVersion: authorization.k8s.io/v1
+kind: SubjectAccessReview
+spec:
+  # requester is the referenced service-account in executor subject
+  user: "system:serviceaccount:<sa namespace>:<sa name>"
+  groups: ["system:serviceaccounts", "system:authenticated", "system:serviceaccounts:<sa namespace>"]
+  # requesting resource attributes
+  resourceAttributes:
+    # verb
+    verb: <the intended verb, can be create/update/patch/delete>
+    # the GVR tuple
+    group: <api group>
+    version: <api version>
+    resource: <api resource name>
+    # the object identifier
+    name: <resource name>
+    namespace: <resource namespace if applicable>
+```
+
+### Architecture
+
+The overall architecture is shown in the picture below:
+
+![Architecture](./arch.png "Architecture")
+
+#### Workflow
+
+The following is what will happen when a `ManifestWork` with non-nil executor
+subject specified is applied to the hub cluster:
+
+
+##### Create/Update sequence
+
+0. Create a "ManifestWork" with non-nil executor subject
+1. The work-agent observes the new resource via watch event
+2. The work-agent builds a SubjectAccessReview request which:
+   - The target subject is the executor subject from `ManifestWork`
+   - The target attribute should contain the "create", "update", "patch", 
+     "get", "list" verbs for all the inlined objects in the spec of 
+     `ManifestWork`. Note that the "resourceNames" should also be specified. 
+3. The work-agent writes the objects to the managed cluster if SAR passes.
+   3.1 Otherwise no objects will be written to the managed cluster and set the 
+       failure condition to the `ManifestWork`.
+4. The work-agent ensures a finalizer upon the `ManifestWork`   
+5. The work-agent reports the results back to the hub cluster.
+   
+
+##### Delete sequence
+
+1. Delete a "ManifestWork" with non-nil executor subject
+2. The work-agent builds a SubjectAccessReview request which:
+    - The target subject is the executor subject from `ManifestWork`
+    - The target attribute should be the "delete" verbs for all the inlined 
+      objects in the spec of `ManifestWork` with  "resourceNames" specified.
```

Comment:
nit: this will need us to return a condition about deletion of manifestwork. We haven't done't this yet.

Reply:
@qiujian16 do mean we can report the deletion failure as a condition in the work's status? currently we have the finalizers, a remaining finalizer on the deleted object indicates that something is failed/unfinished during the background deletion